### ISSUE TITLE
Model misclassifies neutral sentences as positive

### ISSUE THREAD

**Issue by monster0318** [monster0318/sentence-classification#4]
The classifier tends to predict a **positive** label for sentences that are actually **neutral**, especially for inputs with polite or factual tone (e.g., “The meeting was held at 2 PM.”). This indicates potential class imbalance, missing neutral examples, or overfitting during training.

---

**Steps to Reproduce:**

1. Run the model with input: `"The document was submitted on time."`
2. Observe the output label: `positive`
3. Ground truth should be: `neutral`

---

**Expected Behavior:**
Neutral sentences without emotional tone should be labeled as `neutral`, not `positive` or `negative`.

---

**Possible Causes:**

* Training dataset may lack sufficient neutral examples
* Class weights or loss function may not be balanced
* Overfitting to positively skewed training data

---

**Suggested Fixes:**

🔹 **Data Balance Check (Python):**

```python
from collections import Counter
print(Counter([label for _, label in train_data]))
```

🔹 **Model Adjustments:**

* Add more neutral examples to training set
* Use **class weights** in loss function (e.g., `CrossEntropyLoss(weight=...)`)
* Evaluate using a **confusion matrix** to spot class drift

🔹 **Example Fix for PyTorch:**

```python
weights = torch.tensor([1.0, 1.5, 1.0])  # adjust as needed
criterion = nn.CrossEntropyLoss(weight=weights)
```

---
